From a429a6a9c2bcc6996eb530bd6712a02b3bc1fbd8 Mon Sep 17 00:00:00 2001
From: Arulselvan M <arulselvan.m@intel.com>
Date: Fri, 31 May 2013 00:09:32 +0530
Subject: [PORT FROM MAIN]NXP Premium: Integrating NXP Premium library in Middleware

BZ: 59799

Changes are done for integrating NXP premium library for new audio effects

Change-Id: Ia6b96e97c70450417b7d3558b401ad80e685c2fe
Orig-Change-Id: I991cffe31f69fd3a575e778d039538ecd3af0657
Signed-off-by: Arulselvan M <arulselvan.m@intel.com>
---
 media/libeffects/lvm/lib/Android.mk          |    2 ++
 media/libeffects/lvm/wrapper/Android.mk      |    2 ++
 media/libstagefright/Android.mk              |   24 ++++++++++++++++++++++++
 media/libstagefright/AwesomePlayer.cpp       |   21 +++++++++++++++++++++
 media/libstagefright/include/AwesomePlayer.h |    9 +++++++++
 5 files changed, 58 insertions(+), 0 deletions(-)

diff --git a/media/libeffects/lvm/lib/Android.mk b/media/libeffects/lvm/lib/Android.mk
index bb56c75..fdfa429 100644
--- a/media/libeffects/lvm/lib/Android.mk
+++ b/media/libeffects/lvm/lib/Android.mk
@@ -1,3 +1,4 @@
+ifneq ($(strip $(USE_INTEL_LVSE)),true)
 LOCAL_PATH:= $(call my-dir)
 
 # Music bundle
@@ -178,3 +179,4 @@ LOCAL_C_INCLUDES += \
 
 LOCAL_CFLAGS += -fvisibility=hidden
 include $(BUILD_STATIC_LIBRARY)
+endif
diff --git a/media/libeffects/lvm/wrapper/Android.mk b/media/libeffects/lvm/wrapper/Android.mk
index f1af389..554223e 100644
--- a/media/libeffects/lvm/wrapper/Android.mk
+++ b/media/libeffects/lvm/wrapper/Android.mk
@@ -1,3 +1,4 @@
+ifneq ($(strip $(USE_INTEL_LVSE)),true)
 LOCAL_PATH:= $(call my-dir)
 
 # music bundle wrapper
@@ -57,3 +58,4 @@ LOCAL_C_INCLUDES += \
     $(call include-path-for, audio-effects)
 
 include $(BUILD_SHARED_LIBRARY)
+endif
diff --git a/media/libstagefright/Android.mk b/media/libstagefright/Android.mk
index d565660..230bd38 100644
--- a/media/libstagefright/Android.mk
+++ b/media/libstagefright/Android.mk
@@ -1,6 +1,16 @@
 LOCAL_PATH:= $(call my-dir)
 include $(CLEAR_VARS)
 
+ifeq ($(strip $(USE_INTEL_LVSE)),true)
+
+LOCAL_CFLAGS += -DLVSE
+LOCAL_PATH_NXP := ../../../../device/intel/PRIVATE/lifevibes/lvse
+LOCAL_PREBUILT_LIBS += $(LOCAL_PATH_NXP)/libmusicbundle.a
+include $(BUILD_MULTI_PREBUILT)
+endif
+
+include $(CLEAR_VARS)
+
 include frameworks/av/media/libstagefright/codecs/common/Config.mk
 
 LOCAL_SRC_FILES:=                         \
@@ -181,6 +191,20 @@ LOCAL_SHARED_LIBRARIES += \
 
 LOCAL_CFLAGS += -Wno-multichar
 
+ifeq ($(strip $(USE_INTEL_LVSE)),true)
+
+LOCAL_STATIC_LIBRARIES += \
+        libmusicbundle \
+        libLVAudioSource \
+
+endif
+
+ifeq ($(TARGET_HAS_MULTIPLE_DISPLAY),true)
+    LOCAL_C_INCLUDES += $(TARGET_OUT_HEADERS)/display
+    LOCAL_SHARED_LIBRARIES += libmultidisplay
+    LOCAL_CFLAGS += -DTARGET_HAS_MULTIPLE_DISPLAY
+endif
+
 LOCAL_MODULE:= libstagefright
 
 LOCAL_MODULE_TAGS := optional
diff --git a/media/libstagefright/AwesomePlayer.cpp b/media/libstagefright/AwesomePlayer.cpp
index ea6bdbb..36981db 100644
--- a/media/libstagefright/AwesomePlayer.cpp
+++ b/media/libstagefright/AwesomePlayer.cpp
@@ -543,6 +543,12 @@ void AwesomePlayer::reset_l() {
         // _it_ is stopped. Otherwise this is still our responsibility.
         mAudioSource->stop();
     }
+
+#ifdef LVSE
+    LOGV("mLVAudioSource.clear");
+    mLVAudioSource.clear();
+#endif
+
     mAudioSource.clear();
 
     mTimeSource = NULL;
@@ -1452,6 +1458,21 @@ status_t AwesomePlayer::initAudioDecoder() {
 
     if (mAudioSource != NULL) {
         int64_t durationUs;
+
+#ifdef LVSE
+        // insert LifeVibes component
+
+        int32_t sampleRate;
+        (mAudioSource->getFormat())->findInt32(kKeySampleRate, &sampleRate);
+        LOGV("\tLVSE: mAudioSource sampleRate = %d", sampleRate);
+
+        (mAudioTrack->getFormat())->findInt32(kKeySampleRate, &sampleRate);
+        LOGV("\tLVSE: mAudioTrack sampleRate = %d", sampleRate);
+
+        mLVAudioSource = new LVAudioSource(mAudioSource, mAudioSink->getSessionId());
+        mAudioSource = mLVAudioSource;
+#endif
+
         if (mAudioTrack->getFormat()->findInt64(kKeyDuration, &durationUs)) {
             Mutex::Autolock autoLock(mMiscStateLock);
             if (mDurationUs < 0 || durationUs > mDurationUs) {
diff --git a/media/libstagefright/include/AwesomePlayer.h b/media/libstagefright/include/AwesomePlayer.h
index 2306f31..db5d8f0 100644
--- a/media/libstagefright/include/AwesomePlayer.h
+++ b/media/libstagefright/include/AwesomePlayer.h
@@ -21,6 +21,10 @@
 #include "HTTPBase.h"
 #include "TimedEventQueue.h"
 
+#ifdef LVSE
+#include "LVAudioSource.h"
+#endif
+
 #include <media/MediaPlayerInterface.h>
 #include <media/stagefright/DataSource.h>
 #include <media/stagefright/OMXClient.h>
@@ -341,6 +345,11 @@ private:
 
     AwesomePlayer(const AwesomePlayer &);
     AwesomePlayer &operator=(const AwesomePlayer &);
+
+#ifdef LVSE
+    sp<LVAudioSource> mLVAudioSource;
+#endif
+
 };
 
 }  // namespace android
-- 
1.7.4.1

